src/coreclr/gc/generation_fields.h

```diff
@@ -0,0 +1,7 @@
+DEFINE_FIELD      (allocation_context, gc_alloc_context)
```

Comment:
nit - instead of naming these headers generation_fields.h/gcheap_fields.h, I would name them dac_generation_fields.h/dac_gcheap_fields.h, just to make it very clear that these are different from other headers in the sense that they are only used for dac (I'd be fine if you want to make a new dir for them too as we might have others in the future but I don't have a strong opinion as long as it's clear they are for dac).